src/pure/pure.ml

```diff
@@ -31,8 +31,13 @@ let parse_text : state -> string -> string -> Command.t list * state =
   try
     Time.restore t;
     let ast =
-      if old_syntax then Legacy_parser.parse_string fname s
-      else Parser.parse_string fname s
+      let strm =
+        if old_syntax then Parser.Dk.parse_string fname s
+        else Parser.parse_string fname s
+      in
+      let cmds = Stdlib.ref [] in
+      Stream.iter (fun c -> Stdlib.(cmds := c :: !cmds)) strm;
+      List.rev Stdlib.(!cmds)
```

Comment:
I do not remember if you already answered me: can we avoid this List.rev by having a parser entry for a list of commands? If not, what is the problem?

Reply:
A parser for a list of commands cannot be trivially made, it raises an end of stream conflict.

Reply:
Ok. Could you please add a comment about that in the code so that we remember why there is a List.rev?